### ISSUE TITLE
Remove ffast-math

### ISSUE THREAD

**Issue by isuruf** [symengine/symengine#1432]
From what I have seen ffast-math is helpful only if you have floating point arithmetic in your code. (loops with floats, assume commutativity for cse, etc) Since symengine is a symbolic library and not a numerical library there's no advantage. The few places we have floating points is `RealDouble`, `eval_double` etc, but here there are no optimization opportunities since we wrap doubles in `RealDouble` class.

Removal of `ffast-math` would enable us to detect infinities, nans in RealDouble class because with `-ffast-math` the compiler is free to assume that there are no infinites nor nans.

**Comment by isuruf** [comment]
Benchmarks make no difference at all with and without `ffast-math`.

**Comment by certik** [comment]
I am fine with that, if benchmarks are not affected. In my numerical codes, `-ffast-math` provides significant speedup, so I have to use it. But if it makes no difference, we don't need to use it.

**Comment by isuruf** [comment]
> In my numerical codes, -ffast-math provides significant speedup, so I have to use it.

Yes, in numerical codes it will have a significant speedup, but in code like ours, no difference.

> But if it makes no difference, we don't need to use it.

I'll send a PR

**Comment by isuruf** [comment]
Another consequence is https://github.com/symengine/symengine/issues/1453 where subnormal numbers are affected in other libraries.
